crates/sui-framework/sources/crypto/elliptic_curve.move

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+/// Library for Elliptic Curve operations on chain. We specifically support the Ristretto-255 sub-group.
+module sui::elliptic_curve {
+    use std::vector;
+
+    ///////////////////////////////////
+    /// Elliptic Curve structs
+    ///////////////////////////////////
+
+    /// Represents a point on the Ristretto-255 subgroup.
+    struct RistrettoPoint has copy, drop, store {
+        // A 32-byte representation of the group element.
+        value: vector<u8>
+    }
+
+    /// Represents a scalar within the Curve25519 prime-order group.
+    struct Scalar has copy, drop, store {
+        // A 32-byte representation of the scalar
+        value: vector<u8>
+    }
+
+    ///////////////////////////////////
+    /// Private 
+    ///////////////////////////////////
+
+    /// @param value: The value to commit to
+    /// @param blinding_factor: A random number used to ensure that the commitment is hiding.
+    native fun native_create_pedersen_commitment(value: vector<u8>, blinding_factor: vector<u8>): vector<u8>;
```

Comment:
I don't think this should be available on-chain, we can start with no apis that reveal blinding factors.

Reply:
We need this to open commitments on-chain though.

Reply:
makes sense